### ISSUE TITLE
Unexpected scroll occurs when measure is called

### ISSUE THREAD

**Issue by austincondiff** [tannerlinsley/react-virtual#241]
I am using `react-virtual` with `react-table`. When resizing a column, in a `useEffect` I call `measure` (not sure if this is the most efficient way to do this) so that the row can change in size if text within a cell wraps like so...

```js
const parentRef = React.useRef();

const { measure, totalSize, virtualItems } = useVirtual({
  size: rows.length,
  overscan: 50,
  parentRef
});

const columnWidths = state.columnResizing.columnWidths;

useEffect(() => measure(), [columnWidths, measure]);
```

As you can see in the [code sandbox](https://codesandbox.io/s/react-virtualized-resizable-table-spg90?file=/src/App.js:3629-4092), as you resize, the table body scroll position scrolls up each time the measure function is run. This is definitely not expected behavior even if `measure` is just called once.

**Comment by piecyk** [comment]
Yes, `measure` is not efficient way, as it will clear the whole cache.

You can use react ref caveats, for measureRef read size on every re-redner ( it will force browser to reflow on same time ). Remember to add estimateSize, and to be as close to real value as possible, this will will be compared to measure size . Also keyExtractor is very important in situation like sort. Checkout the https://codesandbox.io/s/react-virtualized-resizable-table-forked-1i1tq?file=/src/App.js

**Comment by austincondiff** [comment]
Thank you @piecyk. You have no idea how long I've spent trying to fix this one issue. Your solution works wonderfully! I figured the way I was doing it was inefficient and I read through the documentation but somehow it never occured to me that is how that worked. I really appreciate you pointing me in the right direction. We may consider clearer documentation around `keyExtractor` and how `react-virtual` works with it to expose `virtualItem.key` on each item and `estimatedSize` to calculate heights.
